### PR TITLE
.Net: Fix bug where redis score was mapped from wrong score field.

### PR DESCRIPTION
**Pull request by westey-m**
### Motivation and Context

#9900

### Description

Fixing mapping issue.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Review by dmytrostruk — APPROVED** [review]

**Review by SergeyMenshykh — APPROVED** [review]

File: dotnet/src/Connectors/Connectors.Memory.Redis/RedisHashSetVectorStoreRecordCollection.cs [hunk 1]
```diff
@@ -119,14 +119,14 @@ public RedisHashSetVectorStoreRecordCollection(IDatabase database, string collec
         this._propertyReader.VerifyVectorProperties(s_supportedVectorTypes);
 
         // Lookup storage property names.
-        this._dataStoragePropertyNames = this._propertyReader
-            .DataPropertyStoragePropertyNames
-            .ToArray();
+        var dataStoragePropertyNames = this._propertyReader.DataPropertyStoragePropertyNames;
```
Comment (markwallace-microsoft) [inline dotnet/src/Connectors/Connectors.Memory.Redis/RedisHashSetVectorStoreRecordCollection.cs:122]: Is `this._dataStoragePropertyNames` no longer needed?

File: dotnet/src/Connectors/Connectors.Memory.Redis/RedisVectorStoreCollectionCreateMapping.cs [hunk 2]
```diff
@@ -177,6 +177,7 @@ public static string GetSDKDistanceAlgorithm(VectorStoreRecordVectorProperty vec
         return vectorProperty.DistanceFunction switch
         {
             DistanceFunction.CosineSimilarity => "COSINE",
+            DistanceFunction.CosineDistance => "COSINE",
```
Comment (markwallace-microsoft) [inline dotnet/src/Connectors/Connectors.Memory.Redis/RedisVectorStoreCollectionCreateMapping.cs:180]: These are both the same value, is this correct?

File: dotnet/src/Connectors/Connectors.Memory.Redis/RedisVectorStoreCollectionSearchMapping.cs [hunk 3]
```diff
@@ -117,6 +117,53 @@ public static string BuildFilter(VectorSearchFilter? basicVectorSearchFilter, IR
...
+    {
+        if (redisScore is null)
+        {
+            return null;
+        }
+
+        return distanceFunction switch
+        {
+            DistanceFunction.CosineSimilarity => 1 - redisScore,
+            DistanceFunction.CosineDistance => redisScore,
```
Comment (markwallace-microsoft) [inline dotnet/src/Connectors/Connectors.Memory.Redis/RedisVectorStoreCollectionSearchMapping.cs:160]: Is this reachable?

File: dotnet/src/Connectors/Connectors.Memory.Redis/RedisHashSetVectorStoreRecordCollection.cs (hunk 1 above)
Comment (westey-m) [reply dotnet/src/Connectors/Connectors.Memory.Redis/RedisHashSetVectorStoreRecordCollection.cs:122]: Yeah, let me get rid of it.

File: dotnet/src/Connectors/Connectors.Memory.Redis/RedisVectorStoreCollectionCreateMapping.cs (hunk 2 above)
Comment (westey-m) [reply dotnet/src/Connectors/Connectors.Memory.Redis/RedisVectorStoreCollectionCreateMapping.cs:180]: Yes, redis only support distance, but I am mapping back to similarity if needed on response.

File: dotnet/src/Connectors/Connectors.Memory.Redis/RedisVectorStoreCollectionSearchMapping.cs (hunk 3 above)
Comment (westey-m) [reply dotnet/src/Connectors/Connectors.Memory.Redis/RedisVectorStoreCollectionSearchMapping.cs:160]: Yes, redis will use cosine distance for both and then we will just convert the scores to similarity when responding to the caller.

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**